sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/aad/properties/AadAuthenticationProperties.java

```diff
@@ -126,6 +130,18 @@ public class AadAuthenticationProperties implements InitializingBean {
      */
     private AadApplicationType applicationType;
 
+    private static final Map<AadApplicationType, List<AadAuthorizationGrantType>> compatibleApplicationTypeAndGrantTypes = new HashMap<>();
+
+    static {
+        initCompatibleApplicationTypeAndGrantTypes();
+    }
+
+    private static void initCompatibleApplicationTypeAndGrantTypes() {
+        compatibleApplicationTypeAndGrantTypes.put(WEB_APPLICATION, Arrays.asList(ON_BEHALF_OF));
+        compatibleApplicationTypeAndGrantTypes.put(RESOURCE_SERVER, Arrays.asList(AUTHORIZATION_CODE, ON_BEHALF_OF));
+        compatibleApplicationTypeAndGrantTypes.put(RESOURCE_SERVER_WITH_OBO, Arrays.asList(AUTHORIZATION_CODE));
+    }
```

Comment:
I took another look at the original code, seems like these are incompatible application type and grant types?

Reply:
nonCompatibleApplicationTypeAndGrantTypes will be used